R/Analyses.R

```diff
@@ -210,19 +210,19 @@ createOutcome <- function(outcomeId,
                           riskWindowEnd = NULL,
                           endAnchor = NULL) {
   errorMessages <- checkmate::makeAssertCollection()
-  checkmate::assertInt(outcomeId, add = errorMessages)
+  checkmate::assertNumeric(outcomeId, add = errorMessages)
+  checkmate::assertTRUE(all(outcomeId %% 1 == 0), add = errorMessages)
   checkmate::assertLogical(outcomeOfInterest, add = errorMessages)
   checkmate::assertNumeric(trueEffectSize, len = 1, null.ok = TRUE, add = errorMessages)
   checkmate::assertInt(riskWindowStart, null.ok = TRUE, add = errorMessages)
   checkmate::assertInt(riskWindowEnd, null.ok = TRUE, add = errorMessages)
   checkmate::reportAssertions(collection = errorMessages)
   if (!is.null(startAnchor) && !grepl("start$|end$", startAnchor, ignore.case = TRUE)) {
-    stop("startAnchor should have value 'cohort start' or 'cohort end'")
+    stop("startAnchor should have value \'cohort start\' or \'cohort end\'")
```

Comment:
Could you help me understand why you're escaping single quotes? It is not needed when the string uses double quotes